tensorflow-utils/src/main/java/org/tensorflow/util/buffer/impl/raw/UnsafeReference.java

```diff
@@ -0,0 +1,87 @@
+package org.tensorflow.util.buffer.impl.raw;
+
+import java.lang.reflect.Field;
+import sun.misc.Unsafe;
+
+/**
+ * Holds a valid reference to an instance of sun.misc.Unsafe.
+ */
+public final class UnsafeReference {
```

Comment:
Does this class need to be public?

Reply:
Yes, because `tensorflow-core-api` creates an instance of it. But there is nothing that can be done with it other than passing it back to a `*RawDataBuffer`. Its purpose is to guarantee that the reference holder was able to provide a valid instance of `sun.misc.Unsafe`.

Reply:
Could we have a static factory somewhere that deals with that, so it could remain private?

Reply:
That is what I ended up doing in the end, and I'll probably do the same for all other type of buffers